techbuzz/src/main/java/com/sivalabs/techbuzz/posts/web/controllers/AddVoteController.java

```diff
@@ -4,20 +4,30 @@
 import com.sivalabs.techbuzz.config.annotations.CurrentUser;
 import com.sivalabs.techbuzz.config.logging.Loggable;
 import com.sivalabs.techbuzz.posts.domain.dtos.CreateVoteRequest;
+import com.sivalabs.techbuzz.posts.domain.dtos.PostViewDTO;
+import com.sivalabs.techbuzz.posts.domain.mappers.PostMapper;
+import com.sivalabs.techbuzz.posts.domain.models.Post;
 import com.sivalabs.techbuzz.posts.domain.services.PostService;
 import com.sivalabs.techbuzz.users.domain.models.User;
+import jakarta.annotation.Resource;
 import jakarta.validation.Valid;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
 @Controller
 @Loggable
 class AddVoteController {
     private final PostService postService;
 
+    @Resource(name = "postMapper")
```

Comment:
Instead of using `@Resource`, you can declare it as `private final PostMapper postMapper;` and inject it via constructor just like `PostService`.